core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -290,18 +320,20 @@ private long getTotalPageableItems(UserAnonymizedDto userAnonymized, ChronoUnit
 		Interval interval = Interval.createDayInterval(date);
 		Set<BuddyDto> buddies = buddyService.getBuddiesOfUserThatAcceptedSending(userId);
 
+		LocalDate earliestPossibleDate = determineEarliestBuddyStatusChangeDate(buddies);
```

Comment:
How do you know that the buddies collection is not empty here?

Reply:
Good find! In practice, this will never occur, as we will never return a URL to a "day activity overview with buddies" for a date where the user does not have buddies, but still, we should have appropriate handling. I'll add it.